aspnetcore/fundamentals/configuration/index.md

```diff
@@ -1,452 +1,1308 @@
+| [Environment variables Configuration Provider](#environment-variables-configuration-provider) | Environment variables |
+| [File Configuration Provider](#file-configuration-provider) | Files (INI, JSON, XML) |
+| [Key-per-file Configuration Provider](#key-per-file-configuration-provider) | Directory files |
+| [Memory Configuration Provider](#memory-configuration-provider) | In-memory collections |
 
 ::: moniker-end
 
-::: moniker range="<= aspnetcore-1.1"
+::: moniker range="< aspnetcore-2.1"
+
+| Provider | Provides configuration from&hellip; |
+| -------- | ----------------------------------- |
+| [Command-line Configuration Provider](#command-line-configuration-provider) | Command-line parameters |
+| [Environment variables Configuration Provider](#environment-variables-configuration-provider) | Environment variables |
+| [File Configuration Provider](#file-configuration-provider) | Files (INI, JSON, XML) |
+| [Memory Configuration Provider](#memory-configuration-provider) | In-memory collections |
+
+::: moniker-end
 
-Examples provided in this topic rely upon:
+::: moniker range=">= aspnetcore-1.1"
 
-* Setting the base path of the app with [SetBasePath](/dotnet/api/microsoft.extensions.configuration.fileconfigurationextensions.setbasepath). `SetBasePath` is made available to the app by referencing the [Microsoft.Extensions.Configuration.FileExtensions](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.FileExtensions/) package.
-* Resolving sections of configuration files with [GetSection](/dotnet/api/microsoft.extensions.configuration.configurationsection.getsection). `GetSection` is made available to the app by referencing the [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) package.
-* Binding configuration with [Bind](/dotnet/api/microsoft.extensions.configuration.configurationbinder.bind). `Bind` is made available to the app by referencing the [Microsoft.Extensions.Configuration.Binder](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Binder/) package.
+Additional providers described in other topics are shown in the following table.
+
+| Provider | Provides configuration from&hellip; |
+| -------- | ----------------------------------- |
+| [Azure Key Vault Configuration Provider](xref:security/key-vault-configuration) | Azure Key Vault |
 
 ::: moniker-end
 
-## JSON configuration
+You can also create a [Custom configuration provider](#custom-configuration-provider).
+
+Except for using a File Configuration Provider's reload-on-change capability, configuration sources are read in the order that their configuration providers are specified at startup. The configuration providers described in this topic are described in alphabetical order, not in the order that your code may arrange them. Order configuration providers in your code to suit your priorities for the underlying configuration sources.
+
+A typical sequence of configuration providers is:
```

Comment:
I think this is ok ...

1. They're described alphabetically in this topic. Order them to suit your needs.
1. Here's *typical* sequence.

I don't say the sequence is *recommended*. Since this sequence is the same as `CreateDefaultBuilder`, it should be fine to show. It also forms a good anchor for the remark that CL config provider goes last to allow for overriding.

I'll take another look after the commit.